config/bidderinfo.go

```diff
@@ -170,38 +171,66 @@ func (bi BidderInfo) IsEnabled() bool {
 	return !bi.Disabled
 }
 
-// LoadBidderInfo parses all static/bidder-info/{bidder}.yaml files from the file system.
-func LoadBidderInfo(path string) (BidderInfos, error) {
-	bidderConfigs, err := ioutil.ReadDir(path)
+type InfoReader interface {
+	Read() (map[string][]byte, error)
+}
+
+type InfoReaderFromDisk struct {
+	Path string
+}
+
+func (r InfoReaderFromDisk) Read() (map[string][]byte, error) {
+	bidderConfigs, err := ioutil.ReadDir(r.Path)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	infos := BidderInfos{}
-
+	bidderInfos := make(map[string][]byte)
 	for _, bidderConfig := range bidderConfigs {
 		if bidderConfig.IsDir() {
 			continue //ignore directories
 		}
 		fileName := bidderConfig.Name()
-		filePath := fmt.Sprintf("%v/%v", path, fileName)
+		filePath := filepath.Join(r.Path, fileName)
 		data, err := ioutil.ReadFile(filePath)
 		if err != nil {
 			return nil, err
 		}
+		bidderInfos[fileName] = data
+	}
+	return bidderInfos, nil
+}
+
+func LoadBidderInfoFromDisk(path string) (BidderInfos, error) {
+	bidderInfoReader := InfoReaderFromDisk{Path: path}
+	return LoadBidderInfo(bidderInfoReader)
+}
 
+func LoadBidderInfo(reader InfoReader) (BidderInfos, error) {
+	return processBidderInfos(reader, openrtb_ext.NormalizeBidderName)
+}
+
+func processBidderInfos(reader InfoReader, normalizeBidderName func(string) (openrtb_ext.BidderName, bool)) (BidderInfos, error) {
+	bidderConfigs, err := reader.Read()
+	if err != nil {
+		return nil, fmt.Errorf("error loading bidders data")
+	}
+
+	infos := BidderInfos{}
+
+	for fileName, data := range bidderConfigs {
 		info := BidderInfo{}
 		if err := yaml.Unmarshal(data, &info); err != nil {
-			return nil, fmt.Errorf("error parsing yaml for bidder %s: %v", fileName, err)
+			return nil, fmt.Errorf("error parsing config for bidder %s: %v", fileName, err)
 		}
 
 		bidderName := strings.Split(fileName, ".")
 		if len(bidderName) == 2 && bidderName[1] == "yaml" {
-			// all bidder names loaded from the file system should be in lowercase
-			// to match bidder names in Viper config bidder infos
-			// this will prevent from missing bidder config overrides
-			// in func applyBidderInfoConfigOverrides
-			infos[(strings.ToLower(bidderName[0]))] = info
+			normalizedBidderName, bidderNameExists := normalizeBidderName(bidderName[0])
+			if !bidderNameExists {
+				return nil, fmt.Errorf("error parsing config for bidder %s: unknown bidder", fileName)
+			}
```

Comment:
A map check is less expensive than unmarshalling the `yaml` file. A small optimization would occur if we switch the order of the operations:
```
213   func processBidderInfos(reader InfoReader, normalizeBidderName func(string) (openrtb_ext.BidderName, bool)) (BidderInfos, error) {
214       bidderConfigs, err := reader.Read()
215       if err != nil {
216           return nil, fmt.Errorf("error loading bidders data")
217       }
218   
219       infos := BidderInfos{}
220   
221       for fileName, data := range bidderConfigs {
222 -         info := BidderInfo{}
223 -         if err := yaml.Unmarshal(data, &info); err != nil {
224 -             return nil, fmt.Errorf("error parsing config for bidder %s: %v", fileName, err)
225 -         }
226 - 
227           bidderName := strings.Split(fileName, ".")
228           if len(bidderName) == 2 && bidderName[1] == "yaml" {
229               normalizedBidderName, bidderNameExists := normalizeBidderName(bidderName[0])
230               if !bidderNameExists {
231                   return nil, fmt.Errorf("error parsing config for bidder %s: unknown bidder", fileName)
232               }
    + 
    +             info := BidderInfo{}
    +             if err := yaml.Unmarshal(data, &info); err != nil {
    +                 return nil, fmt.Errorf("error parsing config for bidder %s: %v", fileName, err)
    +             }
    + 
233               infos[string(normalizedBidderName)] = info
234           }
235       }
236       return infos, nil
config/bidderinfo.go
```

Reply:
Sounds like a good idea. I would like to mention this code runs once at startup so performance is not a critical concern with this flow.